packages/vscode-extension/src/debug/taskTerminal/utils/devTunnelStateManager.ts

```diff
@@ -0,0 +1,134 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * @author Xiaofu Huang <xiaofhua@microsoft.com>
+ */
+
+import { Mutex, withTimeout } from "async-mutex";
+import * as fs from "fs-extra";
+import * as path from "path";
+import { isFeatureFlagEnabled } from "@microsoft/teamsfx-core";
+import * as globalVariables from "../../../globalVariables";
+import { FeatureFlags } from "../../../utils/commonUtils";
+
+export interface IDevTunnelState {
+  tunnelId?: string;
+  clusterId?: string;
+  sessionId?: string;
+}
+
+export class DevTunnelStateManager {
+  // TODO: use read-write lock
+  private static mutex = withTimeout(new Mutex(), 1000);
+  private readonly devTunnelStateKey = "teamsToolkit:devtunnel";
+  private stateService: IStateService;
+
+  constructor(stateService: IStateService) {
+    this.stateService = stateService;
+  }
+
+  public static create(): DevTunnelStateManager {
+    const stateService = isFeatureFlagEnabled(FeatureFlags.DevTunnelTest)
```

Comment:
Why adding this flag?

Reply:
It is used for UI test to clean the basis tunnel. The `vscode.context.workspaceState` is saved in sqllite db and hard to get in the UI test.

Reply:
Ok, for test only. Make sense.